sim/midas/src/main/cc/emul/simif_emul.cc

```diff
@@ -29,12 +37,89 @@ simif_emul_t::simif_emul_t(const TargetConfig &config,
       fuzz_seed = strtoll(arg.c_str() + 11, NULL, 10);
       fprintf(stderr, "Using custom fuzzer seed: %ld\n", fuzz_seed);
     }
-    if (arg.find("+fastloadmem") == 0) {
-      fastloadmem = true;
-    }
     if (arg.find("+loadmem=") == 0) {
       load_mem_path = arg.c_str() + 9;
     }
+    if (arg.find("+partition-fpga-cnt=") == 0) {
+      fpga_cnt = atoi(arg.c_str() + 20);
+    }
+    if (arg.find("+partition-fpga-idx=") == 0) {
+      fpga_idx = atoi(arg.c_str() + 20);
+    }
+    if (arg.find("+partition-fpga-topo=") == 0) {
+      int fpga_topo_str = atoi(arg.c_str() + 21);
+      if (fpga_topo_str == 1)
+        comb_logic = true;
+      else if (fpga_topo_str == 2)
+        noc_part = true;
+    }
+  }
+
+  for (int i = 0; i < config.qsfp.num_channels; i++) {
+    qsfp.emplace_back(std::make_unique<qsfp_t>(config.qsfp));
+  }
+
+  printf("qsfp_num_chans: %d\n", config.qsfp.num_channels);
+
+  for (int idx = 0; idx < config.qsfp.num_channels; idx++) {
+    char qsfp_owned_name[257];
+    char qsfp_other_name[257];
+    int next_fpga_idx = (fpga_idx + 1) % fpga_cnt;
+    int prev_fpga_idx = (fpga_idx + fpga_cnt - 1) % fpga_cnt;
+
+    printf("noc_part: %d comb_logic: %d\n", noc_part, comb_logic);
+
+    // TODO : add argument in firesim manager
+    if (!noc_part)
+      assert(fpga_cnt <= 3);
```

Comment:
?

Reply:
This is to match the current FPGA QSFP port limitations. If we are not using the NoC partitioning mode, we can't have more than 4 FPGA connected as each FPGA only has 2 QSFP ports.